week1/tuesday/Assignment_1_2_5/main.cpp

```diff
@@ -0,0 +1,133 @@
+/*
+** Simple library management program. You can add, remove, display books
+** and see if there's any left to be loaned.
+*/
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+struct Book
+{
+    std::string title {};
+    std::string author {};
+    long long   isbn {};
+    int         year {};
+    int         quantity {};
+};
+
+std::vector<Book> library;
+
+/*
+** Displays data of book based on it's title
+*/
+void display_book(std::string title)
+{
+    size_t i { 0 };
+    while (i < library.size())
+    {
+        if (title == library[i].title)
+        {
+            std::cout << "Title:\t\t" << library[i].title << '\n';
+            std::cout << "Author:\t\t" << library[i].author << '\n';
+            std::cout << "ISBN:\t\t" << library[i].isbn << '\n';
+            std::cout << "Year:\t\t" << library[i].year << '\n';
+            std::cout << "Quantity:\t" << library[i].quantity << '\n';
+            return ;
+        }
+        ++i;
+    }
+    std::cout << "Book not found!\n";
+}
+
+/*
+** Adds book to vector
+*/
+void add_book(Book book)
+{
+    library.push_back(book);
+    library.shrink_to_fit();
```

Comment:
What's the purpose of shrinking here? You're only creating an extra action, with no imminent gains? Or am I missing something here...?

Reply:
Because if you shrink after each add, the vector has to reallocate capacity at beginning of each add, which will cause overhead.